src/uct/cuda/base/cuda_md.c

```diff
@@ -35,6 +37,11 @@ ucs_status_t uct_cuda_base_detect_memory_type(uct_md_h md, void *addr, size_t le
             *mem_type_p = UCS_MEMORY_TYPE_CUDA_MANAGED;
         } else {
             *mem_type_p = UCS_MEMORY_TYPE_CUDA;
+            cu_err = cuPointerSetAttribute(&value, CU_POINTER_ATTRIBUTE_SYNC_MEMOPS,
+                                           (CUdeviceptr)addr);
+            if (cu_err != CUDA_SUCCESS) {
+                return UCS_ERR_INVALID_ADDR;
```

Comment:
Its already detected as a CUDA memory. It l might be valid addr, just not able to set the attribute
IMHO, we should warn the user and return the correct mem type(cuda)

Reply:
If we fail in setting attribute, then the application could see a data validation issue later because of the conditions described [here](https://github.com/openucx/ucx/pull/4188#issue-317120563), right? Should we still just issue a warning?

Also, I went with the INVALID_ADDR error because like you mention, pointer is detected as device pointer so this should not fail with any error that it can potentially return (https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__UNIFIED.html#group__CUDA__UNIFIED_1g89f7ad29a657e574fdea2624b74d138e) except possibly something catastrophic. 

What are your thoughts?

Reply:
warning sounds better - the operation (=memory detection) was successful, but something may not work down the road (=potential data corruption)

Reply:
@bureddy and @yosefe 
I've addressed it with a warning. Lmk if other changes look good.